x/derivatives/types/positions.go

```diff
@@ -15,16 +15,38 @@ type PositionInstance interface {
 
 type Positions []Position
 
-func (m Position) Validate() error {
-	if m.RemainingMargin.Amount.IsZero() {
-		return fmt.Errorf("remaining margin cannot be zero")
+func (m Position) IsValid() error {
+	if !m.IsValidMarginAsset() {
+		return fmt.Errorf("margin asset is not valid")
 	}
-	if m.RemainingMargin.Amount.IsNegative() {
-		return fmt.Errorf("remaining margin cannot be negative")
+
+	// check the least requirement for the margin
+	if !m.RemainingMargin.Amount.IsPositive() {
+		return fmt.Errorf("remaining margin must be positive")
+	}
+
+	pfPosition, err := NewPerpetualFuturesPositionFromPosition(m)
+	if err != nil {
+		return err
+	}
+
+	if !pfPosition.IsValidPositionSize() {
+		return fmt.Errorf("position size is not valid")
 	}
+
 	return nil
 }
 
+func (m Position) IsValidMarginAsset() bool {
+	return !(m.Market.BaseDenom != m.RemainingMargin.Denom && m.Market.QuoteDenom != m.RemainingMargin.Denom)
```

Comment:
Oh plz wait.
@mkXultra Is this ok?

Reply:
@taiki1frsh 
is this need condition?
```
m.Market.QuoteDenom != m.RemainingMargin.Denom
```

Reply:
@mkXultra 
Yes, as far as I know.
I fixed the expression and added a comment above.
Plz check it again.